test/v1alpha1/service.go

```diff
@@ -131,6 +132,66 @@ func CreateRunLatestServiceReady(t pkgTest.TLegacy, clients *test.Clients, names
 	return resources, err
 }
 
+// CreateRunLatestServiceReadyWithNumPods creates a new Service in state 'Ready' with the provided number of pods running.
+// This function expects Service and Image name passed in through 'names'. Names is updated with the Route and
+// Configuration created by the Service and ResourceObjects is returned with the Service, Route, and Configuration objects.
+// Returns error if the service does not come up correctly.
+func CreateRunLatestServiceReadyWithNumPods(t pkgTest.TLegacy, clients *test.Clients, names *test.ResourceNames, numPods int, fopt ...rtesting.ServiceOption) (*ResourceObjects, error) {
```

Comment:
This seems quite specific to the test and probably can live in the test file.
I'd also call `numPods` `wantPods`.